src/storage/CookiesStorage.js

```diff
@@ -67,21 +67,67 @@ class CookiesStorage {
     if (document.cookie === oldCookies) {
       return null;
     }
-    
+
     // Since the cookie was updated with a dummy value, remove it by setting
     // expiry to a date in the past.
-    const pastDate = "Thu, 01 Jan 1970 00:00:00 GMT";
-    document.cookie = 
+    const pastDate = 'Thu, 01 Jan 1970 00:00:00 GMT';
+    document.cookie =
       `registrable_domain=;domain=${domain};expires=${pastDate}`;
 
     return domain;
   }
 
   /** @override */
-  save(key, value, secondsToLive) {}
+  save(key, value, secondsToLive =
+    /** @type {number} */ (this.settings_['expires'])) {
+    const valString = JSON.stringify(value);
+    let newCookie = `${this.settings_['prefix']}${key}=${valString}`;
+    newCookie = this.addSettings(newCookie, secondsToLive);
+    document.cookie = newCookie;
+  }
+
+  /**
+   * Finds the registrable domain of current URL and sets it as the default
+   *    domain for cookies.
+   * @param {string} cookie The cookie to add settings to.
+   * @param {?number} secondsToLive The number of seconds cookie will be in
+   *    storage.
+   * @return {string} The cookie with all added settings.
+   * @private
+   */
+  addSettings(cookie, secondsToLive) {
+    cookie = `${cookie}; domain=${this.settings_['domain']}`;
+
+    const expiry = new Date();
+
+    if (secondsToLive === Number.POSITIVE_INFINITY) {
+      const tenYears = 10 * 365 * 24 * 60 * 60;
+      expiry.setMilliseconds(expiry.getMilliseconds() + (tenYears) *
+      1000);
+    } else {
+      expiry.setMilliseconds(expiry.getMilliseconds() + (secondsToLive + 1) *
+      1000);
+    }
+
+    cookie = `${cookie}; expires=${expiry.toUTCString()}`;
+
+    cookie = `${cookie}; ${this.settings_['flags']}`;
+
+    return cookie;
+  }
 
   /** @override */
-  load(key, defaultValue) {}
+  load(key, defaultValue) {
+    if (!document.cookie.includes(`${key}=`)) {
```

Comment:
Includes isn't supported in IE
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

so either it will fail or we'll incur a polyfill